app/controllers/api/v1/merchants_controller.rb

```diff
@@ -0,0 +1,27 @@
+class Api::V1::MerchantsController < ApplicationController
+  def index
+    render json: Merchant.all
+  end
+
+  def show
+    render json: Merchant.find(params[:id])
+  end
+
+  def create
+    render json: Merchant.create!(merchant_params)
+  end
+
+  def update
+    render json: Merchant.update(params[:id], merchant_params)
```

Comment:
This is a much cleaner way to write this, maybe apply this to items and see if that works.